src/wasm/wasm-type.cpp

```diff
@@ -380,7 +380,8 @@ unsigned Type::getByteSize() const {
 }
 
 Type Type::reinterpret() const {
-  auto singleType = *(*this).begin();
+  assert(!isTuple() && "Unexpected tuple type");
+  auto& singleType = *(*this).begin();
```

Comment:
Is there a benefit in making these `const auto&`, incl. `for (const auto& t : type)`?. It's already inferred as `const wasm::Type&` this way, but a quick search also finds opinions preferring to use `const auto` to be sure.

Reply:
Yes, I think so. It's more self-documenting and obvious to the reader that nothing will be mutated that way.

Reply:
Sounds good. Shall I do the `for (const auto& t: type)` changes (that would have been part of https://github.com/WebAssembly/binaryen/pull/3061) in a separate PR to not clutter this one?

Reply:
Took the liberty to outsource: https://github.com/WebAssembly/binaryen/pull/3064